spec/index.html

```diff
@@ -10424,7 +10424,205 @@ <h3>Evaluation Semantics</h3>
+                  <p>
+                    During <a href="#sparqlQuery">translation to the SPARQL algebra</a>
+                  </p>
+                  <pre>
+Replace each occurence of `Y` in X where `Y` is one of
+          <a href="#sparqlTranslateBasicGraphPatterns">Basic Graph Pattern</a>,
+          <a href="#sparqlTranslatePathExpressions">Property Path Expression</a>,
+          <a href="#sparqlTranslateGraphPatterns">`Graph(Var, pattern)`</a>,
+          <a href="#https://www.w3.org/TR/sparql12-query/#sparqlTranslateGraphPatterns">Inline Data</a>
+          with `join(Y, BindingInScope())`.</pre>
+                </div>
+                <div class="note">
+                  c.f. section <a href="#sparqlTranslateGraphPatterns">Translate Graph Patterns</a>
+                  where an empty basic graph pattern start any 
+                  <a href="#rGroupGraphPattern">GroupGraphPattern</a>.
+                  It happens before the <a href="#sparqlSimplification">simplification step</a>.
+                </div>
+              </div>
+
+              <div class="example">
+                <p>
+                  Examples
+                </p>
+              </div>
+
+              <p>
+                Alternative 2: rewrite the algebra during execution. This corresponds to the 
+                <a href="https://github.com/w3c/sparql-dev/blob/main/SEP/SEP-0007/sep-0007.md">original SEP-0007 proposal</a>.
+              </p>
+              <div class="defn">
+                <div>
+                  <b>Definition: <span id="defn_valuesinsertion" name="defn_valuesinsertion">Values Insertion</span></b>
+                  <p>
+                    Define the Values Insertion function `ValuesInsert(X, μ)`
+                  </p>
+                  <pre>Let Table(μ) = { μ } and multiplicity( μ | Table(μ) = { μ } ) = 1
+
+Replace each occurence of `Y` in X where `Y` is one of
+          <a href="#sparqlTranslateBasicGraphPatterns">Basic Graph Pattern</a>,
+          <a href="#sparqlTranslatePathExpressions">Property Path Expression</a>,
+          <a href="#sparqlTranslateGraphPatterns">`Graph(Var, pattern)`</a>,
+          <a href="#https://www.w3.org/TR/sparql12-query/#sparqlTranslateGraphPatterns">Inline Data</a>
+with `join(Y, Table(μ))`.</pre>
+
+                </div>
+              </div>
+            </section>
+
+            <section>
+              <h4>Evaluation of EXISTS</h4>
+              <div class="defn">
+                <b>Definition: <span id="x-defn_evalExists" name="x-defn_evalExists">Evaluation of Exists</span></b>
+                <p>
+                  Let `μ` be the current solution mapping for a filter, and `X` a graph pattern,
+                      define the Evaluation of Exists `exists(X)`
+                </p>
+                <pre>exists(X) = true 
+              if eval( D(G), ValuesInScope(PrjMap(X)), μ)
+                 is a non-empty solution sequence.
```

Comment:
What is `ValuesInScope( .. )`? Where is this defined?

No matter how it is defined, there is another problem here. The formula uses the [eval](https://www.w3.org/TR/sparql12-query/#defn_eval) function with three arguments: i) `D(G)`, ii) the output of `ValuesInScope( .. )`, whatever that might be, and iii) some solution mapping μ. Yet, [eval](https://www.w3.org/TR/sparql12-query/#defn_eval) is not defined for such arguments; it is defined for two arguments: i) `D(G)` and ii) an algebraic expression.

Reply:
Should be `BindingInScope` from earlier - which needs to be defined. I've added editors notes here and earlier.

I'm wondering if a 3-argument `eval` is the better way to express the steps here but for now I've simply removed the `ValuesInScope`.